src/compiler/tracing.ts

```diff
@@ -22,21 +29,12 @@ namespace ts.tracing {
         [key: string]: string | number | boolean | null | undefined | Args | readonly (string | number | boolean | null | undefined | Args)[];
     };
 
-    /** Starts tracing for the given project (unless the `fs` module is unavailable). */
+    /** Starts tracing for the given project. */
     export function startTracing(tracingMode: Mode, traceDir: string, configFilePath?: string) {
-        Debug.assert(!traceFd, "Tracing already started");
+        Debug.assert(!tracing, "Tracing already started");
 
         if (fs === undefined) {
-            try {
-                fs = require("fs");
-            }
-            catch {
-                fs = false;
-            }
-        }
-
-        if (!fs) {
-            return;
+            fs = require("fs"); // want tracing but no fs => let it throw an error

```

Comment:
Should we throw our own error so that we can point users in the right direction?   (e.g. "Tracing requires `fs`").

Reply:
Added, though I'm not sure that there is any point since getting an error about no `fs` would be pretty self-explanatory--?

Reply:
The absence of `fs` is self-explanatory, but not how it interacts with tracing (IMO).